app/javascript/components/Location.js

```diff
@@ -91,22 +91,14 @@ class Location {
     }
   }
 
-  static backgroundColorRGB = (location, is_highlight) => {
+  static backgroundColorRGB = (location, is_highlight, is_search) => {
     let color = location.background_color;
-    if(is_highlight) {
-      switch(color) {
-        case 'transparent':
-        case 'none':
-        case '':
-          color = location.highlightColor;
-          break;
-        default:
-          color = `#${color}`;
-          break;
-      }
-      return `${color}`;
+    if ( is_search && is_highlight ) {
+      color = location.highlightColor;
```

Comment:
If I'm understanding the update here, you want the background highlight colors only to appear when there is a highlight set and if the location was a search result. 

The switch case should still be included here then.